blog/2016-12-05-easy-upgrades-processed-by-git.md

```diff
@@ -0,0 +1,125 @@
+
+## How does it work?
+
+The key step of the process is generating a Git patch. The patch contains all the changes made in the React Native templates between the version your app is using and the new one.
+
+To obtain this patch, we need to generate an app from the templates embedded in the `react-native` package inside your `node_modules` directory the same way the `react-native init` and `react-native upgrade` commands do. Then, after the native apps have been generated from the templates in both the current version and the requested version, Git is able to produce a patch that is adapted to your project (i.e. with your app name):
+
+```
+[...]
+
+diff --git a/ios/MyAwesomeApp/Info.plist b/ios/MyAwesomeApp/Info.plist
+index e98ebb0..2fb6a11 100644
+--- a/ios/MyAwesomeApp/Info.plist
++++ b/ios/MyAwesomeApp/Info.plist
+@@ -45,7 +45,7 @@
+ 		<dict>
+ 			<key>localhost</key>
+ 			<dict>
+-				<key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
++				<key>NSExceptionAllowsInsecureHTTPLoads</key>
+ 				<true/>
+ 			</dict>
+ 		</dict>
+ 		
+[...]
+```
+
+All we need now is to apply this patch to your current source files. While the old `react-native upgrade` process would have prompted you for any difference (even from a single character), Git is able to merge most of the changes automatically using a 3-way merge algorithm and eventually leave us with familiar conflict delimiters:
+
+```
+		13B07F951A680F5B00A75B9A /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+<<<<<<< ours
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/HockeySDK.embeddedframework",
+					"$(PROJECT_DIR)/HockeySDK-iOS/HockeySDK.embeddedframework",
+				);
+=======
+				CURRENT_PROJECT_VERSION = 1;
+>>>>>>> theirs
+				HEADER_SEARCH_PATHS = (
+					"$(inherited)",
+					/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include,
+					"$(SRCROOT)/../node_modules/react-native/React/**",
+					"$(SRCROOT)/../node_modules/react-native-code-push/ios/CodePush/**",
+				);
+```
+
+These conflicts are generally easy to reason about. The delimiter **ours** stands for "your team" whereas **theirs** could be seen as "the React Native team".
+
+## Why introduce a new global package?
+
+React Native comes with a global CLI (the [react-native-cli](https://www.npmjs.com/package/react-native-cli) package) which delegates commands to the local CLI embedded in the `node_modules/react-native/local-cli` directory.
+
+As we mentioned above, the process has to be started from your current React Native version. If we had embedded the implementation in the local-cli, you wouldn't have been able to enjoy this feature until when using old versions of React Native. For example, you wouldn't be able to upgrade from 0.29.2 to 0.38.0 if this new upgrade code was only released in 0.38.0.
```

Comment:
> you wouldn't have been able to enjoy this feature until when using old versions of React Native

I'm not 100% sure but... I understand the opposite with "until". Shouldn't we say "you wouldn't have been able to enjoy this feature when using old versions of React Native" ?